File: tools/bench/answers/go_sonnet/s3/01_rle.go
package main

import "fmt"

func main() {
	s := "aaabbcccca"
	result := ""
	i := 0
	for i < len(s) {
		ch := s[i]
		count := 1
		for i+count < len(s) && s[i+count] == ch {
			count++
		}
		result += fmt.Sprintf("%c%d", ch, count)
		i += count
	}
	fmt.Println(result)
}

File: tools/bench/answers/rust_sonnet/s3/01_rle.rs
fn main() {
    let s = "aaabbcccca";
    let chars: Vec<char> = s.chars().collect();
    let mut result = String::new();
    let mut i = 0;
    while i < chars.len() {
        let c = chars[i];
        let mut count = 1;
        while i + count < chars.len() && chars[i + count] == c {
            count += 1;
        }
        result.push(c);
        result.push_str(&count.to_string());
        i += count;
    }
    println!("{}", result);
}
